File: scripts/export-trades.js
const fs = require('fs')
const path = require('path')
const zlib = require('zlib')

// Parse Excel sheet3 XML
function parseSheet3(buffer) {
  // Find sheet3.xml in the ZIP
  const zipContent = buffer.toString('utf-8')
  const sheet3Marker = 'xl/worksheets/sheet3.xml'
  const sheet3Index = zipContent.indexOf(sheet3Marker)
  
  if (sheet3Index === -1) {
    throw new Error('Sheet3 not found')
  }
  
  // Extract the compressed content (Excel uses DEFLATE)
  // For simplicity, read the whole file as string and parse
  const content = buffer.toString('utf-8', 0, buffer.length)
  
  // Find sheet3 content between markers
  const startMarker = 'xl/worksheets/sheet3.xml'
  const endMarker = 'xl/worksheets/sheet4.xml'
  const startIdx = content.indexOf(startMarker)
  const endIdx = content.indexOf(endMarker, startIdx)
  
  let sheetContent
  if (endIdx > startIdx) {
    sheetContent = content.slice(startIdx + startMarker.length, endIdx)
  } else {
    sheetContent = content.slice(startIdx + startMarker.length)
  }
  
  const trades = []
  
  // Parse rows - look for <row> elements
  const rowRegex = /<row[^\u003e]*>([\s\S]*?)<\/row>/g
  let rowMatch
  let rowNum = 0
  
  while ((rowMatch = rowRegex.exec(sheetContent)) !== null) {
    rowNum++
    if (rowNum === 1) continue // Skip header
    
    const rowXml = rowMatch[1]
    
    // Extract all text content from cells
    // Handle both inline strings and shared strings
    const cellTexts = []
    
    // Match inline strings: <t>TEXT</t>
    const inlineRegex = /<t(?:\s+[^\u003e]*)?>([^\u003c]*)<\/t>/g
    let inlineMatch
    while ((inlineMatch = inlineRegex.exec(rowXml)) !== null) {
      cellTexts.push(inlineMatch[1])
    }
    
    // Match shared string references: <v>INDEX</v>
    const sharedRegex = /<v>(\d+)<\/v>/g
    let sharedMatch
    while ((sharedMatch = sharedRegex.exec(rowXml)) !== null) {
      cellTexts.push(sharedMatch[1])
    }
    
    // We need at least: Trade #, Date, Time, Strategy, Side, Entry, Exit, Status, P&L
    if (cellTexts.length >= 10) {
      const tradeId = parseInt(cellTexts[0]) || 0
      const date = cellTexts[1] || ''
      const time = cellTexts[2] || ''
      const strategy = cellTexts[3] || 'Unknown'
      const side = cellTexts[4] === 'UP' ? 'BUY' : 'SELL'
      const entry = parseFloat(cellTexts[5]) || 0
      const exit = parseFloat(cellTexts[6]) || 0
      const pnlStr = cellTexts[9] || '0' // P&L $ column
      const pnl = parseFloat(pnlStr.replace(/[^\d.-]/g, '')) || 0
      
      if (tradeId > 0 && strategy !== 'Strategy') {
        trades.push({
          id: tradeId,
          time: time || date,
          strategy,
          market: 'BTC-5M',
          side,
          entry,
          exit,
          pnl,
          status: pnl >= 0 ? 'WIN' : 'LOSS'
        })
      }
    }
  }
  
  return trades
}

// Main function
function main() {
  const excelPath = '/root/.openclaw/workspace/polymarket-strategy-tester/live_trading_results.xlsx'
  const outputPath = path.join(__dirname, '..', 'public', 'trades.json')
  
  try {
    if (!fs.existsSync(excelPath)) {
      console.error('Excel file not found:', excelPath)
      process.exit(1)
    }
    
    const buffer = fs.readFileSync(excelPath)
    const trades = parseSheet3(buffer)
    
    // Ensure public directory exists
    const publicDir = path.dirname(outputPath)
    if (!fs.existsSync(publicDir)) {
      fs.mkdirSync(publicDir, { recursive: true })
    }
    
    // Write JSON file
    fs.writeFileSync(outputPath, JSON.stringify({
      trades,
      total: trades.length,
      updated: new Date().toISOString()
    }, null, 2))
    
    console.log(`✓ Exported ${trades.length} trades to ${outputPath}`)
    
    // Log strategy counts
    const strategyCounts = {}
    for (const trade of trades) {
      strategyCounts[trade.strategy] = (strategyCounts[trade.strategy] || 0) + 1
    }
    console.log('\nStrategy breakdown:')
    for (const [strategy, count] of Object.entries(strategyCounts).sort((a, b) => b[1] - a[1])) {
      console.log(`  ${strategy}: ${count}`)
    }
    
  } catch (error) {
    console.error('Error:', error.message)
    process.exit(1)
  }
}

main()

File: scripts/export_trades.py
#!/usr/bin/env python3
"""
Export trades from Excel to JSON for dashboard
"""

import pandas as pd
import json
import os
from datetime import datetime

# Paths
EXCEL_PATH = '/root/.openclaw/workspace/polymarket-strategy-tester/live_trading_results.xlsx'
OUTPUT_PATH = '/root/.openclaw/workspace/herbal_dashboard/public/trades.json'

def main():
    # Read Excel
    df = pd.read_excel(EXCEL_PATH, sheet_name='All Trades')
    
    # Convert to trades format
    trades = []
    for _, row in df.iterrows():
        trade = {
            'id': int(row['Trade #']),
            'time': str(row['Time']),
            'strategy': str(row['Strategy']),
            'market': 'BTC-5M',
            'side': 'BUY' if row['Side'] == 'UP' else 'SELL',
            'entry': float(row['Entry Price']),
            'exit': float(row['Exit Price']) if pd.notna(row['Exit Price']) else 0,
            'pnl': float(row['P&L $']),
            'status': str(row['Status'])
        }
        trades.append(trade)
    
    # Ensure output directory exists
    os.makedirs(os.path.dirname(OUTPUT_PATH), exist_ok=True)
    
    # Write JSON
    output = {
        'trades': trades,
        'total': len(trades),
        'updated': datetime.now().isoformat()
    }
    
    with open(OUTPUT_PATH, 'w') as f:
        json.dump(output, f, indent=2)
    
    print(f"✓ Exported {len(trades)} trades to {OUTPUT_PATH}")
    
    # Strategy breakdown
    strategy_counts = df['Strategy'].value_counts().to_dict()
    print(f"\nStrategy breakdown ({len(strategy_counts)} strategies):")
    for strategy, count in sorted(strategy_counts.items(), key=lambda x: -x[1]):
        print(f"  {strategy}: {count}")

if __name__ == '__main__':
    main()
